### PR TITLE
Cherry picking commits that added new CentOS stacks

### PR DESCRIPTION
**Pull request by l0rd**
### What does this PR do?
Add a spring-boot, wildfly and nodejs centos based stacks to branch `openshift-connector`. These are cherry picks of commits that have already been done (nodejs is still under review) on master branch.

Related to https://github.com/redhat-developer/che-dockerfiles/pull/6/commits/6b66247b8d231b2e2760645ac20db19e9c49337e

### CODE REVIEW

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 1]
```diff
@@ -2140,7 +2140,7 @@
     "creator": "Gytis Trikleris",
     "name": "Spring Boot",
     "description": "Spring Boot Stack on CentOS.",
-    "scope": "advanced",
+    "scope": "global",
```
Comment (ibuziuk) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json]: Why not general ?

**Comment by ibuziuk** [comment]
@l0rd looks good but there are two comments:
- ip validation is failing 
- I think that "global" scope should be chnaged to "general"

**Comment by ibuziuk** [comment]
Actually, while verifying spring-boot stack I got the permission issue: 
directory '//.ssh/': Permission denied:
```
[dev-machine] [STDOUT] Terminal Agent will be downloaded from Workspace Master
[dev-machine] [STDERR] mkdir: cannot create directory '//che': Permission denied
[dev-machine] [STDERR] 
[dev-machine] [STDERR] gzip: stdin: unexpected end of file
[dev-machine] [STDERR] tar: Child returned status 1
[dev-machine] [STDERR] tar: Error is not recoverable: exiting now
[dev-machine] [STDERR] /bin/sh: line 177: //che/terminal/che-websocket-terminal: No such file or directory
```

https://snag.gy/I6C1gS.jpg

Maybe image is not patched on dockerhub - https://hub.docker.com/r/rhche/spring-boot/tags/

**Comment by l0rd** [comment]
Thank you @ibuziuk. I've updated stacks.json (s/global/general/), IP validation should work and I've published the images with the correct tag.

**Review by ibuziuk — APPROVED** [review]

**Comment by TylerJewell** [comment]
Added `target/branch` label to this PR.  Going forward, all non-master branch PRs will need this label.  Please apply going forward.

**Pull request merged**